cli-plugins/socket/socket_darwin.go

```diff
@@ -14,6 +14,6 @@ func listen(socketname string) (*net.UnixListener, error) {
 	})
 }
 
-func onAccept(conn *net.UnixConn, listener *net.UnixListener) {
+func unlink(listener *net.UnixListener) {
 	syscall.Unlink(listener.Addr().String())
```

Comment:
This is somewhat 'optional' in that it's a best-effort attempt to prevent leaking socket names into the filesystem on applicable systems (see the above comment); there are indeed a [litany](https://man7.org/linux/man-pages/man2/unlink.2.html#ERRORS) of possible errors during `unlink(2)`; however given we just created the socket above, it's rather unlikely that this fails.

In the event it does, it's not the end of the world; just a 'leaked name' in the filesystem, in TMPDIR.

Reply:
To please the linters, I think it would be good to `_ = syscall....` here. It also makes it explicit that we chose to ignore it (and not that we overlooked error-handling).

(Bonus points for a short `// comment` that we consider it "best effort", and it's not critical)